### PR TITLE
Use attrs for avoiding creation of excess classes by Styled Components

### LINKED ISSUE #5797: Performance: Improve DisplayElement Wrapper Styled Component
## Feature Description

Getting the following console warning sometimes:

```
Over 200 classes were generated for component displayElement__Wrapper with the id of "displayElement__Wrapper-sc-143lj7s-0".
Consider using the attrs method, together with a style object for frequently changed styles.
Example:
  const Component = styled.div.attrs(props => ({
    style: {
      background: props.background,
    },
  }))`width: 100%;`
```

Code in question is here:

https://github.com/google/web-stories-wp/blob/00db4dae0aa1764fce5f9404d39389fabd6d50f8/assets/src/edit-story/components/canvas/displayElement.js#L45-L51

There might be more areas where this could be immproved.

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by miina**
## Relevant Technical Choices
- Uses `attrs` in places where otherwise SC would create hundreds of classes.

## To-do

- [x] Fix Karma

## User-facing changes
N/A

## Testing Instructions
Open the Text Sets panel. Scroll to the end of the text sets. Now filter the text sets, scroll to the end, remove the filter and scroll again. Verify that no warning about 200+ classes is displayed in the console.

### QA

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #5797

### CODE REVIEW

**Comment by swissspidy** [comment]
Yesss 😻 

Does this eliminate all instances of this warning in the app, or are there still some left?

**Comment by miina** [comment]
> Does this eliminate all instances of this warning in the app, or are there still some left?

I tested the Page Templates and the Text Sets tabs and didn't see more warnings there -- these seem to be the main potential candidates for the warnings. Also added a bunch of elements to the page, duplicated the pages (to see if the carousel would create some excess classes). Seems to be without warnings.

Let me know if you've noticed these anywhere else.

File: assets/src/edit-story/elements/text/display.js [hunk 1]
```diff
@@ -97,11 +98,40 @@ const ForegroundSpan = styled(Span)`
...
+  })
+)`
   margin: 0;
+  position: absolute;
+  z-index: 1;
+  white-space: pre-wrap;
+  overflow-wrap: break-word;
+  word-break: break-word;
+  letter-spacing: normal;
+  color: #000000;
```
Comment (spacedmonkey) [inline assets/src/edit-story/elements/text/display.js]: Hard coded color?

**Review by spacedmonkey — APPROVED** [review]
This looks great, one question.

File: assets/src/edit-story/elements/text/display.js (hunk 1 above)
Comment (miina) [reply assets/src/edit-story/elements/text/display.js]: Thanks for noticing, fixed.

**Comment by miina** [comment]
@swissspidy FYI It looks like I'm going to have to revert this one partially -- it's causing a lot of issues for the Text element editing since we're relying on being able to change the element style directly for various transformations and changes that are all connected.

It probably makes sense to do this change for the Library panel only since there are no transformations happening there. A good issue for looking into this would perhaps be #7637 (for the text sets) -- since this will separate rendering on canvas vs in the library.

**Pull request merged**